wip-trampoline.md

```diff
@@ -1,60 +1,40 @@
 # Trampoline Onion Routing
 
-This document describes the additions we make to the existing onion routing
-to enable the use of trampoline hops.
+This document describes the additions we make to the existing onion routing to
+enable the use of trampoline hops.
 
 ## Packet Structure
 
 ### Trampoline Onion
 
-We define a fixed-size trampoline onion packet that can be nested inside onion
-packets. We define `N` to be the maximum number of trampoline nodes (excluding
-the sender node and the destination node).
+The trampoline onion packet uses the following constants:
 
-The trampoline packet always contains `2*N*FRAME_SIZE + 66` bytes.
+* `FRAME_SIZE_V1` is set to `80` bytes
+* `MAX_HOPS_V1` is set to `8`, allowing up to 8 trampoline hops without
+counting the _origin node_ and _final node_, thus up to 7 _intermediate nodes_
+
+The overall structure of the packet is as follows:
 
 1. type: `trampoline_onion_packet`
 2. data:
-   * [`1`:`version`]
+   * [`1`:`0x01`] (`version`)
    * [`33`:`public_key`]
-   * [`2*N*FRAME_SIZE`:`trampoline_hops_data`]
+   * [`MAX_HOPS_V1*FRAME_SIZE_V1`:`trampoline_hops_data`]
    * [`32`:`hmac`]
 
-Note: we use the same onion format as the standard onion which wastes frames,
-but that theoretically allows another level of nesting (mind blown).
-However not really usable given the space constraints...
-
-TODO: maybe this should be version 1 of the overall onion version namespace?
-It can then also be used as a top-level onion (smaller than the v0 onion).
-It's like the v0 onion but lets each hop choose how they route to the next hop.
-If they have a direct channel they directly send the unwrapped v1 onion to the
-next hop.
-If they don't they wrap the v1 onion inside a v0 to get to the next hop via
-intermediate hops.
-Even though the onion packet is smaller, you don't know if previous hops used
-the direct route or went through v0 wrappings, so you have even greater doubts
-about your position in the route than with only v0.
-
-TODO: maybe we can do something more clever with TLV? At the cost of a second
-onion parser (trampoline onion parser)? Use the top-level version in the parser
-to discriminate?
-
-TODO: version maybe redundant with the TLV type that will include this?
-But keeping it allows using this packet to be used as a top-level onion packet
-where we basically let each hop choose which channel to use to route to the
-next hop. Is this useful?
+This results in a `706`-bytes fixed-size packet.
```

Comment:
Just as a side-note: this leaves us with 594 bytes for the remaining hops, or a maximum routing distance of 9 between two consecutive trampolines.

Reply:
It also means that the maximum route length that is actually enforced by the protocol is `9 * 8 = 72`, which gives us a 72x multiplier for griefing attacks...